pkg/infrastructure/azure/compute.go

```diff
@@ -0,0 +1,221 @@
+	GalleryName          string
+	GalleryImageName     string
+	Region               string
+	Tags                 map[string]*string
+	TokenCredential      azcore.TokenCredential
+	CloudConfiguration   cloud.Configuration
+	OSType               armcompute.OperatingSystemTypes
+	OSState              armcompute.OperatingSystemStateTypes
+	HyperVGeneration     armcompute.HyperVGeneration
+	Publisher            string
+	Offer                string
+	SKU                  string
+	ComputeClientFactory *armcompute.ClientFactory
+}
+
+// CreateGalleryImageOutput contains the return values after creating a gallery
+// image.
+type CreateGalleryImageOutput struct {
+	GalleryImage *armcompute.GalleryImage
+}
+
+// CreateGalleryImage creates a gallery image
+func CreateGalleryImage(ctx context.Context, in *CreateGalleryImageInput) (*CreateGalleryImageOutput, error) {
+	galleryImagesClient := in.ComputeClientFactory.NewGalleryImagesClient()
+
+	galleryImagesPoller, err := galleryImagesClient.BeginCreateOrUpdate(
+		ctx,
+		in.ResourceGroupName,
+		in.GalleryName,
+		in.GalleryImageName,
+		armcompute.GalleryImage{
+			Location: to.Ptr(in.Region),
+			Properties: &armcompute.GalleryImageProperties{
+				OSType:           to.Ptr(in.OSType),
+				OSState:          to.Ptr(in.OSState),
+				HyperVGeneration: to.Ptr(in.HyperVGeneration),
+				Identifier: &armcompute.GalleryImageIdentifier{
+					Publisher: to.Ptr(in.Publisher),
+					Offer:     to.Ptr(in.Offer),
+					SKU:       to.Ptr(in.SKU),
+				},
+			},
+			Tags: in.Tags,
+		},
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create gallery image %s: %w", in.GalleryImageName, err)
+	}
+
+	galleryImagesPollDone, err := galleryImagesPoller.PollUntilDone(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to finish creating gallery image %s: %w", in.GalleryImageName, err)
+	}
+	galleryImage := galleryImagesPollDone.GalleryImage
+	logrus.Infof("GalleryImage.ID=%s", *galleryImage.ID)
+
+	return &CreateGalleryImageOutput{
+		GalleryImage: to.Ptr(galleryImage),
+	}, nil
+
+}
+
+// CreateGalleryImageVersionInput contains the input paramters for creating a
+// gallery image version.
+type CreateGalleryImageVersionInput struct {
+	ResourceGroupName       string
+	GalleryName             string
+	GalleryImageName        string
+	GalleryImageVersionName string
+	Region                  string
+	StorageAccountID        string
+	BlobURL                 string
+	Tags                    map[string]*string
+	RegionalReplicaCount    int32
+	ComputeClientFactory    *armcompute.ClientFactory
+}
+
+// CreateGalleryImageVersionOutput contains the return values after create a
+// gallery image version.
+type CreateGalleryImageVersionOutput struct {
+	GalleryImageVersion *armcompute.GalleryImageVersion
+}
+
+// CreateGalleryImageVersion creates a gallery image version
+func CreateGalleryImageVersion(ctx context.Context, in *CreateGalleryImageVersionInput) (*CreateGalleryImageVersionOutput, error) {
```

Comment:
As these functions can take a while, it would be good to have info level debugging statements at the beginning and ends

Reply:
I've added debug logging